newIDE/app/src/AssetStore/BehaviorStore/BehaviorListItem.js

```diff
@@ -0,0 +1,160 @@
+// @flow
+import * as React from 'react';
+import { type BehaviorShortHeader } from '../../Utils/GDevelopServices/Extension';
+import ButtonBase from '@material-ui/core/ButtonBase';
+import Text from '../../UI/Text';
+import { Trans } from '@lingui/macro';
+import { Column } from '../../UI/Grid';
+import { IconContainer } from '../../UI/IconContainer';
+import HighlightedText from '../../UI/Search/HighlightedText';
+import { type SearchMatch } from '../../UI/Search/UseSearchStructuredItem';
+import Chip from '../../UI/Chip';
+import { LineStackLayout } from '../../UI/Layout';
+import { type SearchableBehaviorMetadata } from './BehaviorStoreContext';
+
+const styles = {
+  button: { width: '100%' },
+  container: {
+    display: 'flex',
+    textAlign: 'left',
+    overflow: 'hidden',
+    width: '100%',
+  },
+  icon: {
+    paddingTop: 4,
+  },
+};
+
+type Props = {|
+  id?: string,
+  objectType: string,
+  objectBehaviorsTypes: Array<string>,
+  behaviorShortHeader: BehaviorShortHeader | SearchableBehaviorMetadata,
+  matches: ?Array<SearchMatch>,
+  onChoose: () => void,
+  onHeightComputed: number => void,
+|};
+
+export const BehaviorListItem = ({
+  id,
+  objectType,
+  objectBehaviorsTypes,
+  behaviorShortHeader,
+  matches,
+  onChoose,
+  onHeightComputed,
+}: Props) => {
+  const alreadyAdded = objectBehaviorsTypes.includes(behaviorShortHeader.type);
+  // An empty object type means the base object, i.e: any object.
+  const isObjectIncompatible =
+    behaviorShortHeader.objectType &&
+    objectType !== behaviorShortHeader.objectType;
+
+  // Report the height of the item once it's known.
+  const containerRef = React.useRef<?HTMLDivElement>(null);
+  React.useLayoutEffect(() => {
+    if (containerRef.current)
+      onHeightComputed(containerRef.current.getBoundingClientRect().height);
+  });
+
+  const renderField = (field: 'description' | 'fullName') => {
+    const originalField = behaviorShortHeader[field];
+
+    if (!matches) return originalField;
+    const nameMatches = matches.filter(match => match.key === field);
+    if (nameMatches.length === 0) return originalField;
+
+    return (
+      <HighlightedText
+        text={originalField}
+        matchesCoordinates={nameMatches[0].indices}
+      />
+    );
+  };
+
+  const isEnabled = !alreadyAdded && !isObjectIncompatible;
```

Comment:
While unusal, should we allow a behavior already added to an object to be added again?